content/react/concepts/hooks/terms/useState/useState.md

```diff
@@ -0,0 +1,90 @@
+---
+Title: 'useState()'
+Description: 'Returns the current state of the component and its setter function'
+Subjects:
+  - 'Web Development'
+Tags:
+  - 'Components'
+  - 'React'
+CatalogContent:
+  - 'react-101'
+  - 'paths/front-end-engineer-career-path'
+---
+
+The **`useState()`** hook returns the current [state](https://www.codecademy.com/resources/docs/react/state) of the [component](https://www.codecademy.com/resources/docs/react/components) and its setter function. State must be changed through the setter only and not be mutated directly as it may cause unexpected behavior.
+
+## Syntax
+
+The `useState()` hook accepts an initial value as its parameter. If no value is passed, it will be `undefined`.
+
+The return value is an array whose first and second elements are the current state and state setter respectively. Those variables can be obtained through array destructuring.
+
+```pseudo
+import React, { useState } from 'react';
+
+const [state, setState] = useState(initialValue);
+```
+
+The `useState()` is imported from the `"react"` library.
+When the component is first rendered, its `state` is assigned an `initialValue` through the `useState()` hook. If one is not provided, the default value is `undefined`. Afterwards, the `setState` function can change the value of `state` and trigger a re-render of the component.
```

Comment:
```suggestion
When the component is first rendered, its `state` is assigned an `initialValue` through the `useState()` hook. If one is not provided, the default value is `undefined`. Afterwards, the `setState` function can change the value of `state` and trigger a re-render of the component.

The `state` must be changed only through the `setState` function and not be mutated directly as this may cause unexpected behavior.
```